src/validation.cpp

```diff
@@ -1975,6 +1976,16 @@ bool CChainState::ConnectBlock(const CBlock& block, BlockValidationState& state,
     int64_t nTime6 = GetTimeMicros(); nTimeCallbacks += nTime6 - nTime5;
     LogPrint(BCLog::BENCH, "    - Callbacks: %.2fms [%.2fs (%.2fms/blk)]\n", MILLI * (nTime6 - nTime5), nTimeCallbacks * MICRO, nTimeCallbacks * MILLI / nBlocksTotal);
 
+    TRACE7(validation, block_connected,
+        block.GetHash().ToString().c_str(),
```

Comment:
It's probably fine in this case, but we should avoid passing formatted data to USDTs when we can format it on the bpf side for performance reasons, especially in tight loops. It looks like we're already passing `GetHash().data()`, can we format this on the ebpf side? I remember having trouble doing this in bpftrace, but perhaps that's a temporary issue until bpftrace gets better buffer formatting.

Reply:
Agree, ideally we'd only pass the `GetHash().data()` as first argument.

>  It looks like we're already passing GetHash().data(), can we format this on the ebpf side? I remember having trouble doing this in bpftrace, but perhaps that's a temporary issue until bpftrace gets better buffer formatting.

Right, the current `%r` printf formatting for buffers in bpftrace isn't suitable for printing header hashes and txids. Adding a new formatting function to bfptrace similar to [this](https://github.com/iovisor/bpftrace/pull/1107/commits/043975141c341d2f64beaef4a99b5f2c0fbfed16#diff-e5786da510ae17c4e629faa5583e79a5f37636702844b066763fba287aa44317R829-R845)  as added in https://github.com/iovisor/bpftrace/pull/1107 doesn't look to hard. However, the block hashes and txids would still be in little-endian.